src/app/pages/system/general-settings/ntp-server/ntp-server-list/ntp-server-list.component.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef,
+} from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { untilDestroyed, UntilDestroy } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { filter } from 'rxjs/operators';
+import { NtpServer } from 'app/interfaces/ntp-server.interface';
+import { EmptyConfig, EmptyType } from 'app/pages/common/entity/entity-empty/entity-empty.component';
+import { NtpServerFormComponent } from 'app/pages/system/general-settings/ntp-server/ntp-server-form/ntp-server-form.component';
+import { WebSocketService, DialogService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-ntp-server-list',
+  templateUrl: './ntp-server-list.component.html',
+  styleUrls: ['./ntp-server-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NtpServerListComponent implements OnInit {
+  dataSource: MatTableDataSource<NtpServer> = new MatTableDataSource([]);
+  displayedColumns = [
+    'address',
+    'burst',
+    'iburst',
+    'prefer',
+    'minpoll',
+    'maxpoll',
+    'actions',
+  ];
+  loading = true;
+  loadingConf: EmptyConfig = {
+    type: EmptyType.Loading,
+    large: false,
+    title: this.translate.instant('Loading...'),
+  };
+  emptyConf: EmptyConfig = {
+    type: EmptyType.NoPageData,
+    large: true,
+    title: this.translate.instant('No servers have been added yet'),
+  };
+  errorConf: EmptyConfig = {
+    type: EmptyType.Errors,
+    large: true,
+    title: this.translate.instant('Can not retrieve response'),
```

Comment:
We can probably have a more generic function to handle errors (and show exactly what the error was). I'll create a ticket for that.

Reply:
Yeah, good idea 👍🏻